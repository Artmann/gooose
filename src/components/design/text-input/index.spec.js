import React from 'react';
import TextInput from '.';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('TextInput', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TextInput label="Name" name="name" value="Chris" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(<TextInput
        label="Name"
        name="email"
        autoFocus={true}
        placeholder="sam@company.com"
        required={true}
        type="email"
        value="chris@company.com"
        />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Triggers onChange', () => {
    const callback = jest.fn();
    const event = { target: { value: 'the-value' } };
    const component = shallow(<TextInput label="Name" name="name" value="Chris" onChange={callback}/>);


    component.find('[data-test-input="name"]').simulate('change', event);

    expect(callback).toHaveBeenCalledWith(event);
  });
});
