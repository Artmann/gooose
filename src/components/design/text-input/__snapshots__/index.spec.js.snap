// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<div
  className="sc-bwzfXH fltgEW"
>
  <label
    className="sc-bdVaJa dGGcPj"
  >
    Name
  </label>
  <input
    autoFocus={false}
    className="sc-htpNat fNIrHl"
    data-test-input="name"
    name="name"
    onChange={[Function]}
    placeholder=""
    required={false}
    type="text"
    value="Chris"
  />
</div>
`;

exports[`TextInput renders correctly with props 1`] = `
<div
  className="sc-bwzfXH fltgEW"
>
  <label
    className="sc-bdVaJa dGGcPj"
  >
    Name
  </label>
  <input
    autoFocus={true}
    className="sc-htpNat fNIrHl"
    data-test-input="email"
    name="email"
    onChange={[Function]}
    placeholder="sam@company.com"
    required={true}
    type="email"
    value="chris@company.com"
  />
</div>
`;
